scala/private/phases/api.bzl

```diff
@@ -22,18 +22,19 @@ def _adjust_phases(phases, adjustments):
     # phase_name: the name of the new phase, also used to access phase information
     # phase_function: the function of the new phase
     for (relation, peer_name, phase_name, phase_function) in adjustments:
-        for idx, (needle, _) in enumerate(phases):
-            if relation in ["^", "first"]:
-                phases.insert(0, (phase_name, phase_function))
-            elif relation in ["$", "last"]:
-                phases.append((phase_name, phase_function))
-            elif needle == peer_name:
-                if relation in ["-", "before"]:
-                    phases.insert(idx, (phase_name, phase_function))
-                elif relation in ["+", "after"]:
-                    phases.insert(idx + 1, (phase_name, phase_function))
-                elif relation in ["=", "replace"]:
-                    phases[idx] = (phase_name, phase_function)
+        if relation in ["^", "first"]:
+            phases.insert(0, (phase_name, phase_function))
+        elif relation in ["$", "last"]:
+            phases.append((phase_name, phase_function))
+        else:
+            for idx, (needle, _) in enumerate(phases):
+                if needle == peer_name:
+                    if relation in ["-", "before"]:
+                        phases.insert(idx, (phase_name, phase_function))
+                    elif relation in ["+", "after"]:
+                        phases.insert(idx + 1, (phase_name, phase_function))
+                    elif relation in ["=", "replace"]:
+                        phases[idx] = (phase_name, phase_function)
```

Comment:
How about adding a `break` in `if` statement? No need to iterate further, right?

Reply:
👍🏽 I’ll also add (in a separate PR?) handling of what happens when someone sends a non existent phase (for example if they have a typo). Right now we will silently ignore. I think we need to explicitly fail. I’ll add that as well but I think separately to keep it clean. Wdyt?

Reply:
Sounds like a good idea.